plugins/app-backend/migrations/20211229105307_init.js

```diff
@@ -20,11 +20,21 @@
  * @param {import('knex').Knex} knex
  */
 exports.up = async function up(knex) {
+  const isMySQL = knex.client.config.client.includes('mysql');
   return knex.schema.createTable('static_assets_cache', table => {
     table.comment(
       'A cache of static assets that where previously deployed and may still be lazy-loaded by clients',
     );
-    table.text('path').primary().notNullable().comment('The path of the file');
+    if (!isMySQL) {
+      table
+        .text('path')
+        .primary()
+        .notNullable()
+        .comment('The path of the file');
+    } else {
+      // cannot have a unique key in mysql with a text column greater than 733 bytes
+      table.text('path').notNullable().comment('The path of the file');
```

Comment:
Not sure we can do this. It was marked as primary for a reason - see [the conflict handling here](https://github.com/backstage/backstage/blob/e383e71c00dd302a8760a91de2addeed51a4871d/plugins/app-backend/src/lib/assets/StaticAssetsStore.ts#L111) that makes sure that the database doesn't fill up with duplicates.

Better to, for now, make it not work with super long path names which are unlikely anyway in a prod scenario

Reply:
Mysql will not allow a unique constraint past 254 characters. What we'll do is instead set the unique constraint on the first 254 characters instead of a full unique constraint specifically for mysql. PR is updated to reflect this. Thanks for showing me the conflict handling in the static asset store.

`| static_assets_cache | CREATE TABLE `static_assets_cache` (
  `path` text NOT NULL COMMENT 'The path of the file',
  `last_modified_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT 'Timestamp of when the asset was most recently seen in a deployment',
  `content` blob NOT NULL COMMENT 'The asset content',
  UNIQUE KEY `static_assets_cache_path_idx` (`path`(254)),
  KEY `static_asset_cache_last_modified_at_idx` (`last_modified_at`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci COMMENT='A cache of static assets that where previously deployed and may still be lazy-loaded by clients' |`